src/main/java/seedu/address/model/inventorymodel/InventoryBook.java

```diff
@@ -101,6 +107,64 @@ public void removeItem(Item key) {
         items.remove(key);
     }
 
+    //// undo-redo operations
+
+    /**
+     * Copies the current {@code InventoryBook} in the state list.
+     */
+    public void commit() {
+        assert inventoryBookStatePointer < inventoryBookStateList.size();
+
+        if (!inventoryBookStateList.isEmpty()) {
+            inventoryBookStateList = inventoryBookStateList.subList(0, inventoryBookStatePointer + 1);
+        }
```

Comment:
i think a better condition would be canRedo() instead

Reply:
having it as canRedo optimizes performance as we only really need to do a copy if there exist downstream "commits"

Reply:
Makes sense, I shall change it